walkeeper/src/replication.rs

```diff
@@ -51,23 +51,23 @@ impl ReplicationConn {
     }
 
     /// Handle incoming messages from the network.
-    ///
     /// This is spawned into the background by `handle_start_replication`.
-    ///
     fn background_thread(mut stream_in: impl Read, timeline: Arc<Timeline>) -> Result<()> {
         // Wait for replica's feedback.
-        // We only handle `CopyData` messages. Anything else is ignored.
-        loop {
-            match FeMessage::read(&mut stream_in)? {
-                Some(FeMessage::CopyData(m)) => {
+        while let Some(msg) = FeMessage::read(&mut stream_in)? {
```

Comment:
Hmm, so the problem was that if `FeMessage::read` returned `Ok(None)`, to indicate EOF, this got stuck in the loop calling `FeMessage::read` over and over again. Is that right?

Would be good to mention in the comment above `FeMessage::read` that it returns None for EOF. And explain the problem that this fixes in a few more sentences in the commit message.

Would it be feasible to write a test or this?

Reply:
> Hmm, so the problem was that if FeMessage::read returned Ok(None), to indicate EOF, this got stuck in the loop calling FeMessage::read over and over again. Is that right?

Yes, that's it.

> Would it be feasible to write a test or this?

Okay, so I tried my best, but it feels a bit unwieldy.

> Would be good to mention in the comment above FeMessage::read that it returns None for EOF.

Sure thing. This is what the rendered docs look like (`cargo doc --open`):

![image](https://user-images.githubusercontent.com/3599598/126561587-4d4ad272-7035-44be-935e-d9910af3d907.png)